workspaces/arborist/lib/arborist/reify.js

```diff
@@ -240,9 +240,9 @@ module.exports = cls => class Reifier extends cls {
 
     // Preserve inertness for failed stuff.
     if (this.actualTree) {
-      for (const [loc, actual] of this.actualTree.inventory.entries()) {
+      for (const actual of this.actualTree.inventory.values()) {
         if (actual.ideallyInert) {
-          const ideal = this.idealTree.inventory.get(loc)
+          const ideal = this.idealTree.inventory.get(actual.name)
```

Comment:
I wasn't able to get this working with the loc. Is it possible we wanted the name for the idealTree inventory?

Reply:
that's pretty strange, but something doesn't feel right about using the `actual.name`. Isn't that the *package* name? In that case, this is absolutely not what you want. You want the full path of the dependency in the tree (which I've been assuming is what loc is).

What was wrong with loc? My local testing made everything seem ok.

Reply:
@zkat https://github.com/npm/cli/pull/8184/commits/64b838401a58381e9c9327a6f82bef853a24448a

Okay, I reverted it - but struggling to get the tests set up to find a value there

Reply:
ah hmmm

Reply:
update: sussed this out with @owlstronaut over DMs. :)

tl;dr the failing coverage is because we need a test that covers "install w/ a failing optional, then rerun a 'no-op' install", since that's what'll trigger this block of code that's just meant to preserve the ideallyInert flag in that specific situation.

Reply:
https://github.com/npm/cli/pull/8184/commits/f635c978a3045bcb35322717c656a02e2ceafeb4

Done! Thank you :)